### PR TITLE
Avoid loading ICU data early during mono startup

### PR DESCRIPTION
**Pull request by pavelsavara**
Current mono startup does
`AssemblyNameParser` -> `CreateCultureWithInvariantData` -> `GlobalizationMode.Invariant` -> `GlobalizationMode.Settings.cctor` -> `LoadICU` during startup.

This is attempt to postpone loading ICU until really needed. 
It matters to Browser OS because we could start the runtime before the ICU data are finished downloading.

We could make this conditional for `#if !TARGET_BROWSER` if necessary.

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs [hunk 1]
```diff
@@ -25,6 +25,10 @@ private static partial class Settings
         // This allows for the whole Settings nested class to be trimmed when Invariant=true, and allows for the Settings
         // static cctor (on Unix) to be preserved when Invariant=false.
         internal static bool Invariant => Settings.Invariant;
+
```
Comment (ilonatommy) [inline src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs:28]: What about putting it into `#if TARGET_BROWSER`? Same about changes in `CultureData`

**Comment by pavelsavara** [comment]
Now we need to test/prove this is enough to start runtime without ICU data.

**Comment by pavelsavara** [comment]
Let's continue after https://github.com/dotnet/runtime/pull/99924 landed

There is nice [Log](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-100587-merge-3f8aef5561c74eb398/Wasm.Browser.Threads.Sample/1/console.cb195295.log?helixlogtype=result)
```
[08:18:41] fail: [out of order message from the browser]: http://127.0.0.1:46569/_framework/dotnet.runtime.js 2 [MONO] System.TypeInitializationException: TypeInitialization_Type, <Module>
                  ---> System.TypeInitializationException: TypeInitialization_Type, Settings
                  ---> System.Exception: Pavel: Testing if this will fail during startup or only when in use
                    at System.Globalization.GlobalizationMode.Settings..cctor()
                    Exception_EndOfInnerExceptionStack
                    at System.Globalization.GlobalizationMode.Settings.get_Invariant()
                    at System.Globalization.GlobalizationMode.get_Invariant()
                    at System.Globalization.CultureData.GetNFIValues(NumberFormatInfo )
                    at System.Globalization.NumberFormatInfo..ctor(CultureData )
                    at System.Globalization.CultureInfo.get_NumberFormat()
                    at System.Globalization.NumberFormatInfo.<GetInstance>g__GetProviderNonNull|58_0(IFormatProvider )
                    at System.Globalization.NumberFormatInfo.GetInstance(IFormatProvider )
                    at System.Int32.TryParse(ReadOnlySpan`1 , NumberStyles , IFormatProvider , Int32& )
                    at System.Version.TryParseComponent(ReadOnlySpan`1 , String , Boolean , Int32& )
                    at System.Version.ParseVersion(ReadOnlySpan`1 , Boolean )
                    at System.Version.TryParse(ReadOnlySpan`1 , Version& )
                    at System.Environment.get_Version()
                    at System.Runtime.InteropServices.JavaScript.__GeneratedInitializer.__Net7SelfInit_()
                    at <Module>..cctor()
                    Exception_EndOfInnerExceptionStack
```

cc @kg

**Comment by pavelsavara** [comment]
closing in favor of https://github.com/dotnet/runtime/pull/100141

**Pull request closed without merging**